### PR TITLE
build: update golang to 1.22

### PR DESCRIPTION
**Pull request by tobybellwood**
This PR:
* Updates the version of golang used to 1.22
* Updates the upstream images to Alpine3.20
* Updates the kubectl binary to 1.30 and latest helm binary
* Updates most go deps

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
     - name: Install Go
       uses: actions/setup-go@v4
       with:
-        go-version: "^1.21"
```
Comment (smlx) [inline .github/workflows/build.yaml:16]: Not sure if you are aware but you can use `stable` here to automatically use the latest release. It should be pretty safe due to Go's [compatibility guarantee](https://go.dev/blog/compat).

Comment (tobybellwood) [reply .github/workflows/build.yaml]: we haven't done that, so that we can have a unified strategy - and move all the services to 1.22 at the same time.

**Comment by shreddedbacon** [comment]
What, if any, are the potential impacts of enabling buildkit by default?

**Comment by tobybellwood** [comment]
I _think_ at this stage none, but will obviously have to test thoroughly across a range of scenarios to be sure...

**Review by shreddedbacon — APPROVED** [review]
This seems fine to me

**Pull request merged**